connectors/sources/s3.py

```diff
@@ -126,7 +133,12 @@ async def get_bucket_objects(self, bucket):
                 bucket_obj = await s3.Bucket(bucket)
                 await asyncio.sleep(0)
 
-                async for obj_summary in bucket_obj.objects.page_size(page_size):
+                if "prefix" in kwargs:
```

Comment:
This will be evaluated to `True` when you pass `prefix=""`

Reply:
`prefix=""` is the default behaviour for filtering out the data. So when `prefix=""` that means it will fetch all the object for defined bucket.

Reply:
This will not throw any error as the default value for prefix is `""`. but yes let me change it a bit to make it more clear.